pkg/migration/migrations/fix_privileges_migration.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package migrations
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/migration"
+	"github.com/cockroachdb/cockroach/pkg/security"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
+	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/errors"
+)
+
+func fixPrivilegesMigration(
+	ctx context.Context, _ clusterversion.ClusterVersion, d migration.TenantDeps,
+) error {
+	var lastUpgradedID descpb.ID
+	for {
+		done, idToUpgrade, objectType, err := findNextDescriptorToFix(ctx, d.InternalExecutor, lastUpgradedID)
+		if err != nil || done {
+			return err
+		}
+		if err := fixPrivileges(ctx, idToUpgrade, d, objectType); err != nil {
+			return err
+		}
+		lastUpgradedID = idToUpgrade
+	}
+}
+
+func fixPrivileges(
+	ctx context.Context, upgrade descpb.ID, d migration.TenantDeps, objectType privilege.ObjectType,
+) error {
```

Comment:
This might be out of scope but would it be possible to also apply `descpb.MaybeFixPrivileges` and perhaps `descpb.MaybeFixUsagePrivForTablesAndDBs` in this migration instead of having to rely solely on the `RunPostDeserializationChanges` method of `catalog.DescriptorBuilder` for this purpose?